TravelAgency-service/src/main/java/cz/muni/fi/travelAgency/facade/ExcursionFacadeImpl.java

```diff
@@ -25,17 +28,21 @@
     @Autowired
     private ExcursionService excursionService;
 
+    @Autowired
+    private TripService tripService;
+
     /**
      * Mapper responsible for mapping DTOs to Entities.
      */
     @Autowired
     private BeanMappingService beanMappingService;
 
     @Override
-    public void createExcursion(ExcursionDTO excursionDTO) {
+    public Long createExcursion(ExcursionCreateDTO excursionDTO) {
         Excursion mappedExcursion = beanMappingService.mapTo(excursionDTO, Excursion.class);
-        excursionService.createExcursion(mappedExcursion);
-        excursionDTO.setId(mappedExcursion.getId());
+        mappedExcursion.setTrip(tripService.findById(excursionDTO.getTripId()));
```

Comment:
is this needed, the trip should be mapped by the service. In case it works fine, I don't think setting is needed.

Reply:
From web it could get only id of trip, not the whole object.

Reply:
Sooo why do we fix this and not the web ? :D I feel like this functionality is used by many cases when the object is actually there.